### PR TITLE
Correct pvc storageclass indentation.

### PR DESCRIPTION
**Pull request by HumairAK**
Resolve: #508

### CODE REVIEW

**Comment by dsp-developers** [comment]
A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-509`
An OCP cluster where you are logged in as cluster admin is required.

To use this image run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines-operator.git
cd data-science-pipelines-operator/
git fetch origin pull/509/head
git checkout -b pullrequest 09f0346777c93c1bd75bb54515d13aae17d2974f
make deploy IMG="quay.io/opendatahub/data-science-pipelines-operator:pr-509"
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

**Comment by VaniHaripriya** [comment]
/lgtm

The corrected indentation for storageClassName fixes the below error in DSPO logs:

2023-12-12T17:57:07Z	ERROR	Reconciler error	{"controller": "datasciencepipelinesapplication", "controllerGroup": "datasciencepipelinesapplications.opendatahub.io", "controllerKind": "DataSciencePipelinesApplication", "DataSciencePipelinesApplication": {"name":"sample","namespace":"dspa"}, "namespace": "dspa", "name": "sample", "reconcileID": "2eaa4e56-1f1d-4c46-8599-c1d081c9db52", "error": "error loading template (minio/pvc.yaml.tmpl) yaml: error converting YAML to JSON: yaml: line 11: did not find expected key"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:326
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:273
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.13.0/pkg/internal/controller/controller.go:234

**Review by gmfrasca — APPROVED** [review]

File: config/internal/mariadb/pvc.yaml.tmpl [hunk 1]
```diff
@@ -10,7 +10,7 @@ spec:
   accessModes:
     - ReadWriteOnce
   {{- if .MariaDB.StorageClassName }}
```
Comment (gmfrasca) [inline config/internal/mariadb/pvc.yaml.tmpl:12]: not part of this review but the extraneous `-` at the beginning here feels like it should break something?

Comment (gmfrasca) [reply config/internal/mariadb/pvc.yaml.tmpl:12]: nvm, this is part of trimming in templates.  I'm not entirely sure it's necessary here but not worth blocking the review over

**Pull request merged**